### PR TITLE
feat: add kOps aws support

### PR DESCRIPTION
**Pull request by saumas**

### CODE REVIEW

File: internal/services/providers/kops/kops.go [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+	"castai-agent/internal/config"
+	awsclient "castai-agent/internal/services/providers/eks/client"
+	"castai-agent/internal/services/providers/gke"
+	"castai-agent/internal/services/providers/types"
+	"castai-agent/pkg/labels"
+)
+
+const Name = "kops"
+
+func New(_ context.Context, log logrus.FieldLogger, clientset kubernetes.Interface) (types.Provider, error) {
```
Comment (jmazionis) [inline internal/services/providers/kops/kops.go:27]: context seems to be redundant as param here

File: internal/config/config.go [hunk 2]
```diff
@@ -120,6 +133,21 @@ func Get() Config {
 		}
 	}
 
+	if cfg.KOPS != nil {
+		if cfg.KOPS.CSP == "" {
+			requiredDiscoveryDisabled("KOPS_CSP")
```
Comment (zilvinasu) [inline internal/config/config.go:138]: nitpick, could you rename the func to requiredWhenDiscoveryDisabled ?

File: internal/services/providers/kops/kops.go [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+	}
+
+	return false, nil
+}
+
+func (p *Provider) Name() string {
+	return Name
+}
+
+func (p *Provider) getClusterNameAndStateStore(ns *v1.Namespace) (clusterName, stateStore *string, reterr error) {
```
Comment (zilvinasu) [inline internal/services/providers/kops/kops.go:149]: add comment with example

File: internal/services/providers/kops/kops_test.go [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+		var objects []runtime.Object
+
+		namespaceID := uuid.New()
+
+		objects = append(objects, &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:  k8stypes.UID(namespaceID.String()),
+				Name: metav1.NamespaceSystem,
+				Annotations: map[string]string{
+					"addons.k8s.io/core.addons.k8s.io": `{"version":"1.4.0","channel":"s3://test-kops/test.k8s.local/addons/bootstrap-channel.yaml","manifestHash":"3ffe9ac576f9eec72e2bdfbd2ea17d56d9b17b90"}`,
```
Comment (zilvinasu) [inline internal/services/providers/kops/kops_test.go:47]: does this work the same for older and newer kops versions?

File: internal/services/providers/kops/kops.go [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+	p.csp = csp
+
+	if p.csp == "aws" {
+		opts := []awsclient.Opt{
+			awsclient.WithMetadata("", region, clusterName),
+			awsclient.WithEC2Client(),
+		}
+		c, err := awsclient.New(ctx, p.log, opts...)
+		if err != nil {
+			p.log.Errorf("failed initializing aws client, spot functionality will be reduced: %v", err)
```
Comment (zilvinasu) [inline internal/services/providers/kops/kops.go:87]: `, spot functionality for savings estimation will be reduced`

**Review by zilvinasu — APPROVED** [review]

File: internal/services/providers/kops/kops_test.go (hunk 4 above)
Comment (saumas) [reply internal/services/providers/kops/kops_test.go:47]: Yes! Tested from 1.11 to 1.20.

**Pull request merged**